internal/colerrs/README.md

```diff
@@ -0,0 +1,17 @@
+# Internal collector errors
+
+This package includes helpers propagating errors from exporters or processors
+back to receivers. Error propagation is helpful in a setup where the Collector
+acts as a simple gateway between clients and other Collector instances or
+backends.
+
+Error propagation only works when there's synchronous processing of a pipeline.
+Concretely, this means there should be no batch processors, sending queues for
+exporters and other resiliency mechanisms should be disabled.
+
+"Asymmetrical" processors (e.g. Anything that merges, splits, reorganizes,
```

Comment:
Would this be true also for anything that transforms the data? I'm wondering what would happen if the transform processor is used in a pipeline, and it modified the data in a way that caused the backend to reject it. If the backend returns an error, that's propagated to the client, what can the client do about it?

Reply:
That's a great case to document!